src/FshConversionProvider.ts

```diff
@@ -0,0 +1,182 @@
+import { Uri, TextDocumentContentProvider, EventEmitter, workspace, OutputChannel } from 'vscode';
+import { basename, dirname } from 'path';
+import YAML from 'yaml';
+import { SushiConfiguration } from './utils';
+
+export class FshConversionProvider implements TextDocumentContentProvider {
+  static readonly fshConversionProviderScheme = 'fshfhirconversion';
+  private mapContent = new Map<string, string>();
+
+  private onDidChangeEmitter = new EventEmitter<Uri>();
+  onDidChange = this.onDidChangeEmitter.event;
+
+  dispose(): void {
+    this.onDidChangeEmitter.dispose();
+  }
+
+  updated(newContent: string, newUri: Uri): void {
+    this.mapContent.set(newUri.fsPath, newContent);
+
+    this.onDidChangeEmitter.fire(newUri);
+  }
+
+  provideTextDocumentContent(uri: Uri): string {
+    return this.mapContent.get(uri.fsPath);
+  }
+}
+
+export function createFSHURIfromFileUri(fileUri: Uri, prefix: string): Uri {
+  let fileName = basename(fileUri.path);
+  fileName = fileName.split('.').slice(0, -1).join('.');
+
+  return createURIfromFileUri(fileName, '.fsh', prefix);
+}
+
+export function createJSONURIfromFileUri(identifier: string, prefix: string): Uri {
+  return createURIfromFileUri(identifier, '.json', prefix);
+}
+
+function createURIfromFileUri(identifier: string, extension: string, prefix: string): Uri {
+  //Get the file name without the extension
+  return Uri.parse(
+    FshConversionProvider.fshConversionProviderScheme +
+      ': (PREVIEW) ' +
+      prefix +
+      identifier +
+      extension
+  );
+}
+
+export async function findVersionAnddependencies(
+  fileUri: Uri,
+  output: OutputChannel
+): Promise<{ version: string; dependencies: string[]; sushiconfig: Uri }> {
+  let fhirVersion = '4.0.1';
+  const conversionDependencies: string[] = [];
+
+  const sushiConfigUri = await findMatchingSushiConfig(fileUri);
+  if (sushiConfigUri) {
+    output.appendLine('Sushi-config.yaml file found: ' + sushiConfigUri.fsPath);
+
+    const configContents = await workspace.fs.readFile(sushiConfigUri);
+
+    const decoder = new TextDecoder();
+    const decodedConfig = decoder.decode(configContents);
+    const parsedConfig = YAML.parse(decodedConfig);
+
+    fhirVersion = getVersionFromSushiConfig(parsedConfig);
+    output.appendLine('Found version: ' + fhirVersion);
+
+    const dependencies = getDependenciesFromSushiConfig(parsedConfig);
+    dependencies.forEach(dep => {
+      output.appendLine('Found dependency: ' + dep.packageId + '@' + dep.version);
+      conversionDependencies.push(dep.packageId + '@' + dep.version);
+    });
+  } else {
+    output.appendLine('No sushi-config.yaml file found.');
```

Comment:
In the case of a project w/ FHIR JSON (and not FSH), there probably is not a sushi-config.yaml. Instead, there might be an ImplementationGuide JSON file (e.g., `ImplementationGuide-my.package.id.json`). For example, if you download and unzip a package like https://www.hl7.org/fhir/us/core/package.tgz and then try to convert one of its JSON files.

So... you _could_ consider looking for an ImplementationGuide file if a sushi-config.yaml file is not found. That's up to you. I think we will be very glad to accept this PR without support for ImplementationGuide configs, so if you don't have time (or interest) to do it, that's OK. In that case, however, perhaps you could put a comment here with a todo? (E.g., `// TODO: Look for and use an ImplementationGuide JSON file if there is no sushi-config.yaml`).

Reply:
This is interesting I haven't thought on that one. For now I will add the TODO and then think about adding this feature.